transformer_engine/common/fused_attn/fused_attn_fp16_bf16_max_seqlen_512.cu

```diff
@@ -629,27 +640,32 @@ void fused_attn_max_512_fwd_impl(int64_t b, int64_t h, int64_t s_q, int64_t s_kv
                                  NVTE_QKV_Layout layout, NVTE_Bias_Type bias_type,
                                  NVTE_Mask_Type mask_type, void *devPtrQ, void *devPtrK,
                                  void *devPtrV, void *devPtrS, void *devPtrO, void *devPtrBias,
-                                 void *devCuSeqlenQ, void *devCuSeqlenK, void *workspace,
-                                 size_t *workspace_size, cudnnDataType_t tensorType,
+                                 void *devPtrCuSeqlenQ, void *devPtrCuSeqlenKV,
+                                 void *devPtrDropoutSeed, void *devPtrDropoutOffset,
+                                 void *workspace, size_t *workspace_size,
+                                 cudnnDataType_t tensorType, cudnnDataType_t rng_state_type,
                                  cudaStream_t stream, cudnnHandle_t handle) {
     try {
-        constexpr int64_t seed = 0;  // TODO(rewang): replace this with device seed/offset
         NVTE_CHECK_CUDNN(cudnnSetStream(handle, stream));
 
-        FADescriptor descriptor{b,           h,
-                                s_q,         s_kv,
-                                d,           scaling_factor,
-                                is_training, dropout_probability,
-                                layout,      bias_type,
-                                mask_type,   tensorType};
+        FADescriptor descriptor{b,
+                                h,
+                                s_q,
+                                s_kv,
+                                d,
+                                scaling_factor,
+                                is_training,
+                                dropout_probability,
+                                layout,
+                                bias_type,
+                                mask_type,
+                                tensorType,
+                                rng_state_type};
 
         using CacheType = std::map<FADescriptor, cudnn_frontend::ExecutionPlan>;
         static thread_local CacheType fmha_fprop_cache;
 
-        bool enable_dropout = (dropout_probability != 0.0f);
-
-        NVTE_CHECK(!enable_dropout,
-                   "dropout probability > 0 in fused_attn_max_512 has not been implemented.");
+        bool enable_dropout = (dropout_probability != 0.0f) && (!is_training);
```

Comment:
Shouldn't we only apply dropout during training?
```suggestion
        bool enable_dropout = (dropout_probability != 0.0f) && is_training;
```

Reply:
Yes, it was a bug. I fixed it and add the related tests to cover this in 17a183c73c41c988625ae28c779564d3b5fb2afd. Thanks.